### PR TITLE
pdf: Add binds for `pdf-history-forward/backward`

### PR DESCRIPTION
**Pull request by PRESFIL**
### Brief summary of what the package does

`pdf-tools` has functions for navigating in jump history. This PR simply adds 
standard bindings to these functions. The prefix counters are also supported.

### Direct link to the package repository

https://github.com/vedang/pdf-tools

### Checklist

Assume you're working on `mpc` mode:

- [x] byte-compiles cleanly
- [x] `M-x checkdoc` is happy. Don't manually write `(provide 'evil-collection-mpc)`, `M-x checkdoc` can do it automatically for you
- [x] define `evil-collection-mpc-setup` with `defun`
- [x] define `evil-collection-mpc-mode-maps` with `defconst`
- [x] All functions should start with `evil-collection-mpc-`

### CODE REVIEW

File: modes/pdf/evil-collection-pdf.el [hunk 1]
```diff
@@ -95,6 +99,24 @@ Consider COUNT."
       (image-bob)
       (image-set-window-hscroll hscroll))))
 
+(defun evil-collection-pdf-jump-forward (&optional count)
+  "Wrap `pdf-history-forward' with `evil'.
+
+Consider COUNT."
+  (interactive "P")
+  (if count
```
Comment (condy0919) [inline modes/pdf/evil-collection-pdf.el:107]: `(or count 1)`

File: modes/pdf/evil-collection-pdf.el [hunk 2]
```diff
@@ -95,6 +99,24 @@ Consider COUNT."
...
+      (pdf-history-forward count)
+    (pdf-history-forward 1)))
+
+(defun evil-collection-pdf-jump-backward (&optional count)
+  "Wrap `pdf-history-backward' with `evil'.
+
+Consider COUNT."
+  (interactive "P")
+  (if count
+      (pdf-history-backward count)
```
Comment (condy0919) [inline modes/pdf/evil-collection-pdf.el:117]: `(or count 1)`

File: modes/pdf/evil-collection-pdf.el [hunk 3]
```diff
@@ -195,6 +217,13 @@ Consider COUNT."
   (evil-collection-define-key 'visual 'pdf-view-mode-map
     "y" 'pdf-view-kill-ring-save)
 
+  (evil-collection-inhibit-insert-state 'pdf-history-minor-mode-map)
+  (evil-set-initial-state 'pdf-history-minor-mode 'normal)
+  (evil-collection-define-key 'normal 'pdf-history-minor-mode-map
+    ;; history forward / backward
+    (kbd "C-i") 'evil-collection-pdf-jump-forward
+    (kbd "C-o") 'evil-collection-pdf-jump-backward)
```
Comment (condy0919) [inline modes/pdf/evil-collection-pdf.el:225]: Does it work? Most minors need `evil-normalize-keymaps`. e.g. https://github.com/emacs-evil/evil-collection/blob/master/modes/auto-package-update/evil-collection-auto-package-update.el#L38

**Review by condy0919 — CHANGES_REQUESTED** [review]

File: modes/pdf/evil-collection-pdf.el (hunk 3 above)
Comment (PRESFIL) [reply modes/pdf/evil-collection-pdf.el:225]: It turned out that it really didn't stop working after a clean run. For some reason,
the original version doesn't have these hooks for the other minor modes either
(`pdf-outline-buffer-mode` and `pdf-occur-buffer-mode-map`).
But `pdf-outline-buffer-mode` bindings works...

I can add `evil-normalize-keymaps` if needed, as far as I understand, this should
be done for each minor mode separately.

Comment (PRESFIL) [reply modes/pdf/evil-collection-pdf.el:225]: Sorry for misleading, I confirm that this works the bindings work without
`evil-normalize-keymaps` for `pdf-history-minor-mode`. What confuses me
is that `pdf-tools`'s jump history are page-accurate, not character-accurate
like in Vim. You can see this in the header using the `pdf-history-debug` function.

**Comment by PRESFIL** [comment]
I confirm that this works the bindings work without `evil-normalize-keymaps`
for `pdf-history-minor-mode`. For some reason, the original version doesn't
have these hooks for the other minor modes either
(`pdf-outline-buffer-mode` and `pdf-occur-buffer-mode-map`).

I can add evil-normalize-keymaps if needed, as far as I understand, this should
be done for each minor mode separately.

Other things was resolved.

File: modes/pdf/evil-collection-pdf.el (hunk 3 above)
Comment (condy0919) [reply modes/pdf/evil-collection-pdf.el:223]: OK.

File: modes/pdf/evil-collection-pdf.el [hunk 4]
```diff
@@ -40,12 +42,16 @@
...
+(declare-function pdf-history-forward "pdf-history")
+(declare-function pdf-history-backward "pdf-history")
 
 (defvar pdf-view-mode-map)
+(defvar pdf-history-minor-mode-map)
 (defvar pdf-outline-buffer-mode-map)
 (defvar pdf-occur-buffer-mode-map)
 
 (defvar pdf-view-mode-map)
+(defvar pdf-history-minor-mode-map)
```
Comment (condy0919) [inline modes/pdf/evil-collection-pdf.el:54]: Why there are duplicates? Cleanup.

It comes from this commit.

```diff
commit 323bb7d85848a6a142ae14f39c3a073ce6423e20
Author: James Nguyen <james@jojojames.com>
Date:   Sun Jan 24 00:56:41 2021 -0800

    Add scroll-lock (#358)

diff --git a/modes/pdf/evil-collection-pdf.el b/modes/pdf/evil-collection-pdf.el
--- /dev/null
+++ b/modes/pdf/evil-collection-pdf.el
@@ -0,0 +44,7 @@
+(defvar pdf-view-mode-map)
+(defvar pdf-outline-buffer-mode-map)
+(defvar pdf-occur-buffer-mode-map)
+
+(defvar pdf-view-mode-map)
+(defvar pdf-outline-buffer-mode-map)
+(defvar pdf-occur-buffer-mode-map)
```

And I can confirm we can remove one copy.

**Review by condy0919 — CHANGES_REQUESTED** [review]

**Comment by condy0919** [comment]
Squash yours commits please.

File: modes/pdf/evil-collection-pdf.el (hunk 4 above)
Comment (condy0919) [reply modes/pdf/evil-collection-pdf.el:54]: Sorry, it's from 

```diff
commit a5303b07eedb8bb620c3489b8ee9b27744d49cb7
Author: Pierre Neidhardt <ambrevar@gmail.com>
Date:   Sun Feb 25 16:43:08 2018 +0100

    Silence compiler warnings

diff --git a/evil-collection-pdf.el b/evil-collection-pdf.el
--- a/evil-collection-pdf.el
+++ b/evil-collection-pdf.el
@@ -37,6 +37,10 @@
+(defvar pdf-view-mode-map)
+(defvar pdf-outline-buffer-mode-map)
+(defvar pdf-occur-buffer-mode-map)
+(declare-function evil-collection-inhibit-insert-state "evil-collection")
 
 (declare-function evil-collection-inhibit-insert-state "evil-collection")
 
 (defvar pdf-view-mode-map)
 (defvar pdf-outline-buffer-mode-map)
 (defvar pdf-occur-buffer-mode-map)

commit 5b2da79e7ea708224bb042cf4a45cc1d5397ed59
Author: James Nguyen <james@jojojames.com>
Date:   Sat Feb 24 17:14:05 2018 -0800

    Fix linting errors

diff --git a/evil-collection-pdf.el b/evil-collection-pdf.el
--- a/evil-collection-pdf.el
+++ b/evil-collection-pdf.el
@@ -37,1 +37,6 @@
 
+(declare-function evil-collection-inhibit-insert-state "evil-collection")
+
+(defvar pdf-view-mode-map)
+(defvar pdf-outline-buffer-mode-map)
+(defvar pdf-occur-buffer-mode-map)
```

The 2 commits do the same thing.

**Comment by PRESFIL** [comment]
Squashed.

**Comment by PRESFIL** [comment]
Looks like no my faults ;)

Snapshot CI failed due deprecation in another file:

```
In toplevel form:
modes/emms/evil-collection-emms.el:63:16: Error: ‘point-at-bol’ is an obsolete function (as of 29.1); use ‘line-beginning-position’ instead.
make: *** [Makefile:11: compile] Error 1
Error: Process completed with exit code 2.
```

**Comment by condy0919** [comment]
@tshu-w Help! pdf-tools needs your tests/contribution!

**Comment by tshu-w** [comment]
@condy0919 Hey, I'm not using pdf-tools anymore, but I may test this PR when I'm free if needed.

**Comment by PRESFIL** [comment]
Are we still waiting for the test?

@condy0919, Maybe you can test it yourself?

Installing `pdf-tools` is not that difficult, apart from the package you need to
install AUR's `emacs-pdf-tools-git` package.

**Comment by condy0919** [comment]
Sorry for delay, let me test.

**Comment by PRESFIL** [comment]
Any updates?

**Comment by PRESFIL** [comment]
:rocket:

**Pull request merged**